src/components/content-tab-overview.tsx

```diff
@@ -142,18 +140,10 @@ function CustomizeSection( {
 
 function ShortcutsSection( { selectedSite }: Pick< ContentTabOverviewProps, 'selectedSite' > ) {
 	const { terminalWpCliEnabled } = useFeatureFlags();
-	const { terminal, handleTerminalChange, getSavedTerminal } = useTerminalData();
-	const { editor, handleEditorChange, getSavedEditor } = useEditorData();
+	const { data: editor } = useGetUserEditorQuery();
+	const { data: terminal } = useGetUserTerminalQuery();
 
-	useEffect( () => {
-		handleEditorChange( editor );
-		handleTerminalChange( terminal );
-	}, [ editor, handleEditorChange, handleTerminalChange, terminal ] );
-
-	useIpcListener( 'user-preference-changed', async () => {
-		handleEditorChange( await getSavedEditor() );
-		handleTerminalChange( await getSavedTerminal() );
-	} );
+	const terminalName = supportedTerminalNames[ terminal as keyof typeof supportedTerminalNames ];
```

Comment:
```suggestion
	const terminalName = getTerminalName( terminal );
```
Consider this a suggestion, but a helper function could help us do away with the type casting. This could go in `src/modules/user-settings/lib/terminal.ts`:

```ts
export function getTerminalName( terminal: SupportedTerminal | undefined ): string {
	if ( ! terminal ) {
		return '';
	}

	return supportedTerminalNames[ terminal ];
}
```